src/records.jl

```diff
@@ -97,13 +97,21 @@ function Base.iterate(rec::T, state=0) where T <: AttributeRecord
     return (fieldname(T, state) => get(getfield(rec, state)), state)
 end
 
-"""
-    getattribute(rec::AttributeRecord, attr::Symbol)
-"""
-getattribute(rec::AttributeRecord, attr::Symbol) = getfield(rec, attr)
+function getattribute end  # Remove when depwarn in `getindex` is removed
 
```

Comment:
Don't we want to do

```
Base.getproperty(rec, attr) = get(getfield(rec, attr))

@deprecate getindex(rec, attr) getproperty(rec, attr)
```
?

NOTE: `hasmethod` seems like a slow operation to do in a `getindex` method

Reply:
Maybe? I was going off the original discussion of it here but I might have understood wrong: https://github.com/invenia/Memento.jl/pull/86#discussion_r195262795

Reply:
I have made some changes that would be similar to what Rory suggested that allows tests to pass. Note that this is now a breaking changes PR because of the changes in the behaviour of `getproperty` (see the changes in the tests) and because subtypes of `AttributeRecord` probably don't have the correct `getproperty` behaviour already implemented that the `getindex` deprecation redirects to.

I'm actually not sure how useful deprecating `getindex` is since people were supposed to be implementing `getattribute` and not `getindex` for subtypes of `AttributeRecord`.